pkg/wsnet/net.go

```diff
@@ -24,7 +24,11 @@ func UpsertNetRule(netRules []wsv1alpha1.NetworkRule, r wsv1alpha1.NetworkRule)
 		}
 	}
 	if index >= 0 {
-		netRules[index] = r
```

Comment:
What's the purpose of this change?

Reply:
By current function definition, it is hard for caller to know that TargetPortNumber will not be upserted.

Caller expects pure upsert behavior that just given NetworkRule is upserted.

In the reality, however, TargetPortNumber is not required and only AuthProxy concerns about the spec. So I think it is better TargetPortNumber is a pointer and the expected upsert behavior is like TargetPortNumber will not replaced when it is nil.

Please add the comment of the background and TODO

Reply:
This method is called by cosmoctl and dashboard, and the caller is duplicated, so it will be integrated into one file with the next refactoring.
This method is called by Cosmoctl and Dashboard, and the caller sourcode is duplicated, so it will be integrated into one file.
Net.go also integrates into these, so I want this Pull Request to MERGE.

https://github.com/oruharo/cosmo/blob/wip2/pkg/kosmo/workspace.go#L162-L208